### ISSUE TITLE
An elegent solution for WGAN-gp gradient penalty

### ISSUE THREAD

**Issue by Godricly** [dmlc/mxnet#5982]
Is there any elegant way to compute gradient of WGAN-gp gradient loss term in MXNet?  It seems that I have 2 handcraft it. 

the one in tf looks very convenient.
```python
differences = fake_data - real_data
interpolates = real_data + (alpha*differences)
gradients = tf.gradients(Discriminator(interpolates), [interpolates])[0]
slopes = tf.sqrt(tf.reduce_sum(tf.square(gradients), reduction_indices=[1]))
gradient_penalty = tf.reduce_mean((slopes-1.)**2)
disc_cost += LAMBDA*gradient_penalty
```

**Comment by sxjscience** [comment]
(Point to the discussion here) https://github.com/dmlc/mxnet/issues/5699

**Comment by Godricly** [comment]
closed for duplication.
